core/src/main/java/com/yahoo/oak/Block.java

```diff
@@ -27,7 +22,8 @@
         this.id = NativeMemoryAllocator.INVALID_BLOCK_ID;
         // Pay attention in allocateDirect the data is *zero'd out*
         // which has an overhead in clearing and you end up touching every page
-        this.buffer = ByteBuffer.allocateDirect(this.capacity);
+        this.memAddress = UnsafeUtils.unsafe.allocateMemory(this.capacity);
```

Comment:
Better to create an `allocateMemory()` interface of `UnsafeUtils` and not to access the `unsafe` field. It should be private.